src/core/simulation.jl

```diff
@@ -319,25 +320,27 @@ end
 """ # TODO: Add DocString
 function build!(sim::Simulation; kwargs...)
     check_kwargs(kwargs, SIMULATION_BUILD_KWARGS, "build!")
-    _check_sequence(sim)
-    _check_chronologies(sim)
+    if isnothing(sim.sequence) || isempty(sim.stages)
+        throw(ArgumentError("The simulation object requires a valid definition of stages and SimulationSequence"))
+    end
+    _check_forecasts_sequence(sim)
```

Comment:
Should the call to PYS.check_forecast_consistency below be moved into this function?

Reply:
That's already called when making the PSI container. We would en up calling it twice.